models/Result.js

```diff
@@ -215,45 +239,54 @@ module.exports = class Results {
     const results = [];
     const onePersonResultIds = [];
 
-    client.connect();
-    client.on('error', (err) => {
-      logger.errorLog('Results.switchSpotsForPerformance', err);
-    });
-
-    const resultsQuery = client.query(resultsSql, [id]);
-
-    resultsQuery.on('row', result => results.push(result));
-    resultsQuery.on('end', () => {
-      const filteredResults = results.filter(({ namenumbertwo }, i) => {
-        if (!namenumbertwo) {
-          onePersonResultIds.push(results[i].resultsid);
-        }
-        return namenumbertwo;
+    return new Promise((resolve, reject) => {
+      client.connect();
+      client.on('error', (err) => {
+        logger.errorLog('Results.switchSpotsForPerformance', err);
+        reject(err);
       });
-      const filteredResultsIds = filteredResults.sort(resultsSort).map(({ resultsid }) => resultsid);
 
-      const switchQuery = client.query(switchSql, [filteredResultsIds]);
+      const resultsQuery = client.query(resultsSql, [id]);
 
-      switchQuery.on('end', () => {
-        const oneUserQuery = client.query(oneUserSql, [onePersonResultIds]);
+      resultsQuery.on('row', result => results.push(result));
+      resultsQuery.on('end', () => {
+        const filteredResults = results.filter(({ namenumbertwo }, i) => {
+          if (!namenumbertwo) {
+            onePersonResultIds.push(results[i].resultsid);
+          }
+          return namenumbertwo;
+        });
+        const filteredResultsIds = filteredResults.sort(resultsSort).map(({ resultsid }) => resultsid);
```

Comment:
This bothers me. Make it `{ resultid }`. There's only one result id per index of the array